code/game/objects/structures/teleporter.dm

```diff
@@ -141,17 +160,25 @@
 	///The optional cell to power the teleporter if off the grid
 	var/obj/item/cell/cell
 	COOLDOWN_DECLARE(teleport_cooldown)
+	
+	///tag to avoid mixups. Purely for players, not for systems.
+	var/tele_tag
 
 /obj/item/teleporter_kit/Initialize()
 	. = ..()
 	AddElement(/datum/element/deployable_item, /obj/machinery/deployable/teleporter, type, 2 SECONDS)
 	cell = new /obj/item/cell/high(src)
+	tele_tag = rand(1, 999)
+	name = "\improper ASRS Bluespace teleporter #[tele_tag]"
+	
 
 /obj/item/teleporter_kit/Destroy()
+	linked_teleporter.linked_teleporter = null
```

Comment:
Make sure you're actually linked or this can runtime

Reply:
Good catch.